src/tests.rs

```diff
@@ -191,17 +194,60 @@ mod buf {
             buf.put_prefix_varint(*n);
         }
         let mut result = vec![];
-        let mut decode_data = buf.as_slice();
+        let decode_data = buf.as_slice();
         for decoded in decode_data.iter_prefix_varint::<i16>() {
             result.push(decoded.unwrap());
         }
         assert_eq!(to_encode, result.as_slice());
     }
 
+    #[test]
+    fn iterator_on_vec_works() {
+        let to_encode = [1, 2, -30, -24_000];
+        let mut buf = vec![];
+        for n in to_encode.iter() {
+            buf.put_prefix_varint(*n);
+        }
+        let mut result = vec![];
+        for decoded in buf.iter_prefix_varint::<i16>() {
```

Comment:
result is ~write-only here, maybe instead:
```
assert_eq!(buf.iter_prefix_varint::<16>().collect::<Result<Vec<_>,_>().unwrap().as_slice());
```

Are you still able to to read `buf` after iteration completes, maybe just `assert!(buf.len() >= result.len());`? IIUC this should still be possible but would like to confirm in the test.

Please make similar changes to `iterator_on_vec_deq_works`